### ISSUE TITLE
Let sponsor configure bunq

### ISSUE THREAD

**Issue by jamalv** [teamforus/general#346]
Learn more about change requests here: https://bit.ly/39CWeEE

### Change description
Main CR: As a sponsor I would like to configure bunq so that I can automatically process payments.

Context: As a sponsor I would like to set up a fund without needing foundation Forus for assistance.

MVP: add to implementation dashboard (field to enter bunq key) - There's many ways to improve the MVP; but for now let's keep that out of scope.

Figma: https://www.figma.com/file/YcMvVCqbIYmnrsKFPMfqdA/Untitled?node-id=2%3A1481

**Comment by jamalv** [comment]
@maxvisser could you rate this on scope and check out if the figma prototype is sufficient? If it's small (few hours max) it can be planned in.

**Comment by maxvisser** [comment]
Is this part of this cr? As a sponsor I would like to set up a fund without needing foundation Forus for assistance.

**Comment by maxvisser** [comment]
I would do a functional design first, aswell to look at Email / DigiD settings what there went wrong (why it is now an unreleased feature). The scope currently in figma is small but probably there are some missing links.

**Comment by jamalv** [comment]
> Is this part of this cr? As a sponsor I would like to set up a fund without needing foundation Forus for assistance.

The literal requirement is _"Sponsor kan zelfstandig een fonds aanmaken"_ which translates to: "Sponsor can independently create a fund" - If we add this feature and and article to the helpcenter about opening and connecting funds with bunq; then I think this is possible?

**Comment by jamalv** [comment]
Also is email/DigiD settings unreleased or is it hidden behind a role?

**Comment by maartenfv** [comment]
I really like this issue. It would help with implementation. 

In Figma is see that it would be placed under webshop. I consider this to be a strange place for this input.

Could it be placed under organisation settings? 

![image](https://user-images.githubusercontent.com/30211022/89989682-93fdd800-dc81-11ea-97d0-5d60bfbd41fa.png)

**Comment by jamalv** [comment]
@maartenfv Sure! 😄 

@maxvisser in general I think it makes sense if bunq is connected to the organisation. I don't see any usecases where one organisation would have multiple bunq accounts, what do you think?

**Comment by maxvisser** [comment]
A provider would also be able to add a bunq key; organizations are not role specific.

**Comment by jamalv** [comment]
Jup; understood. 

@maartenfv what do you think about that? 

- If we add it like the original proposal it will only be possible to configure bunq if you also have an implementation (your own webshop)
- If we add it like you proposed, it will be possible for **any** organisation to connect a bunq account 😉 - so also shopkeepers, they cannot really use it, unless they also choose to set up a fund 😛 

I added a new figma protoype for the context you proposed: https://www.figma.com/file/YcMvVCqbIYmnrsKFPMfqdA/Untitled?node-id=11%3A0

**Comment by maxvisser** [comment]
> @maartenfv Sure! 😄
> 
> @maxvisser in general I think it makes sense if bunq is connected to the organisation. I don't see any usecases where one organisation would have multiple bunq accounts, what do you think?

There is more still to do:
- Add fund to implementation (choose which webshop)
- Fund states need to be refactored currently we have four: waiting, pending and active and closed
Waiting: Waiting on configuration
Pending: fund is configured, pending on start date
Active: running till end date
Closed: end_date has arived no new transaction can happen. vouchers expire etc.

So if the sponsor will do all steps themselves we need two more CR requests.

**Comment by jamalv** [comment]
@maartenfv, this means that the scope will be smaller if we add it to the implementation settings (so connect a bunq account for a specific webshop instead of to an organisation) - what do you think? Should we extend the scope and add the other change requests? Or just add it like this?

I would propose to then to add this like the original proposal (to the webshop), to make the upcoming implementations already easier, and to then re-consider this bit when we add PSD2 bank connections.

**Comment by maartenfv** [comment]
@jamalv I think it is better to add it to the organisation. It is oke that providers will see this as well. By doing this we will not have to change it later. Adding it to the webshop will make the system less user friendly.

**Comment by maxvisser** [comment]
maybe make it a seperate entity which a organization can create and manage. So you add a new table to db with monatary accounts. On fund creation you can choose these. Could be just a page on sponsor dashboard.

**Comment by maxvisser** [comment]
A decision needs to be maded on which route to go. After choosing lets plan this issue as it is a Must have.

**Comment by maxvisser** [comment]
Also DigiD and E-mail configuration are still unreleased features. We should see how we can release those aswell and not make the same mistakes here (why those are not released yet)

**Comment by maartenfv** [comment]
@maxvisser Max and I discussed this during a meeting on 30-09. We decided we want to add it to the organisation. However, Max stated this would have implications. I suggest putting this issue in the icebox until we know about BNG. We can discuss this with customers if it poses a problem.

**Comment by jamalv** [comment]
Thanks for the updates @maxvisser and @maartenfv! Updating the status to on-hold and adding it to the icebox, I agree it is better to do the BNG connection first.

**Comment by maxvisser** [comment]
This has been released
